lib/Controller/DelegationController.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * @author Cyrille Bollu <cyr.debian@bollu.be> for Arawa (https://www.arawa.fr/)
+ *
+ * GroupFolders
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\GroupFolders\Controller;
+
+use OCP\IConfig;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\OCSController;
+use OCP\IGroupManager;
+use OCP\IRequest;
+
+class DelegationController extends OCSController {
+
+	/** @var IConfig */
+	private $config;
+
+	/** @var IGroupManager */
+	private $groupManager;
+
+	public function __construct($AppName,
+			IConfig $config,
+			IGroupManager $groupManager,
+			IRequest $request) {
+		parent::__construct($AppName, $request);
+		$this->config = $config;
+		$this->groupManager = $groupManager;
+	}
+
+	/**
+	 * Returns the list of all groups
+	 *
+	 * @NoAdminRequired
+	 * @RequireGroupFolderAdmin
+	 *
+	 * @return DataResponse
+	 */
+	public function getAllGroups() {
+		// Get all groups
+		$groups = $this->groupManager->search('');
+
+		// transform in a format suitable for the app
+		$data = [];
+		foreach($groups as $group) {
+			$data[] = [
+				'id' => $group->getGID(),
+				'displayname' => $group->getDisplayName(),
+				'usercount' => $group->count(),
+				'disabled' => $group->countDisabled(),
+				'canAdd' => $group->canAddUser(),
+				'canRemove' => $group->canRemoveUser(),
+			];
+		}
+
+		// return info
+		return new DataResponse($data);
+	}
+
+	/**
+	 * Get the list of groups allowed to use groupfolders
+	 *
+	 * @NoAdminRequired
+	 * @RequireGroupFolderAdmin
+	 *
+	 * @return DataResponse
+	 */
+	public function getAllowedGroups() {
+		$groups = json_decode($this->config->getAppValue('groupfolders', 'delegated-admins', '["admin"]'));
```

Comment:
I would leave out the admin group in any parts as it should always have access and is also granted that in the DelegationService

Reply:
makes sense. 

Do you want me to also remove the 'admin' group from the list of selectable group in the frontend?

Reply:
Yes, I think it makes sense to exclude that overall and just have it be enabled by default for admins. Then there is no way to lock out real administrators by accident and whenever we change something in the behavior of how system admins are handled, this does not affect this feature.

Reply:
done